app/src/routes/transfer/(components)/transfer-form.svelte

```diff
@@ -212,13 +215,91 @@ const transfer = async () => {
     return toast.error("No evm wallet connected")
   if ($fromChain.rpc_type === "cosmos" && !$userAddress.cosmos)
     return toast.error("No cosmos wallet connected")
+
+  if ($fromChain.rpc_type === "aptos" && !$userAddress.aptos)
+    return toast.error("No aptos wallet connected")
+
   if (!$receiver) return toast.error("Invalid receiver")
 
   let supported = getSupportedAsset($fromChain, $asset.address)
   let decimals = supported?.decimals ?? 0
   let parsedAmount = parseUnits(amount, decimals)
 
-  if ($fromChain.rpc_type === "cosmos") {
+  /** --- APTOS START --- */
+  if ($fromChain?.rpc_type === "aptos") {
+    const { connectedWallet, connectionStatus } = get(aptosStore)
+    if ($userAddressAptos === null) return toast.error("No aptos user address found")
+
+    if (connectionStatus !== "connected" || !connectedWallet) {
+      transferState.set({
+        kind: "SWITCHING_TO_CHAIN",
+        warning: new Error("No wallet connected")
+      })
+      return
+    }
+
+    const wallet = getAptosWallet(connectedWallet)
+    if (!wallet) {
+      transferState.set({
+        kind: "SWITCHING_TO_CHAIN",
+        warning: new Error(`${connectedWallet} wallet not found`)
+      })
+      return
+    }
+
+    // @ts-ignore
+    transferState.set({ kind: "SWITCHING_TO_CHAIN" })
+
+    const rpcUrl = $fromChain?.rpcs.find(rpc => rpc.type === "rpc")?.url
+    if (!rpcUrl) return toast.error(`no rpc available for ${$fromChain?.display_name}`)
+
+    if (stepBefore($transferState, "CONFIRMING_TRANSFER")) {
+      const chainInfo = await wallet.getNetwork()
+
+      if (chainInfo?.chainId.toString() !== $fromChainId) {
+        transferState.set({
+          kind: "SWITCHING_TO_CHAIN",
+          warning: new Error("Failed to switch chain")
+        })
+        return
+      }
+
+      // @ts-ignore
+      transferState.set({ kind: "CONFIRMING_TRANSFER" })
+    }
+
+    if (stepBefore($transferState, "TRANSFERRING")) {
+      try {
+        const client = createUnionClient({
+          chainId: "2",
+          account: await wallet?.getAccount(),
+          transport: wallet as AptosBrowserWallet
+        })
+
+        const transferPayload = {
+          simulate: true,
+          receiver: $receiver,
+          amount: parsedAmount,
+          authAccess: "wallet",
+          denomAddress: $assetAddress,
+          destinationChainId: $toChain.chain_id as ChainId
+        } satisfies TransferAssetsParameters<"2">
+
+        const transfer = await client.transferAsset(transferPayload)
+
+        if (transfer.isErr()) throw transfer.error
+        transferState.set({ kind: "TRANSFERRING", transferHash: transfer.value })
+      } catch (error) {
+        if (error instanceof Error) {
+          // @ts-ignore
+          transferState.set({ kind: "CONFIRMING_TRANSFER", error })
+        }
+        return
+      }
+    }
+
+    /** --- APTOS END --- */
+  } else if ($fromChain.rpc_type === "cosmos") {
```

Comment:
We definitely need to refactor the transfer-form, but I consider that to be out of scope for this PR